Py3 support

peterVorman opened the pull request.

FuhuXia committed:
remove CKAN 2.3 tests and functionality

Commit:
Add CI tests for CKAN 2.8 and 2.9

Commit:
Update README

Commit:
Update docker environment for matrix builds

Support development and testing using CKAN_VERSION make variable. Move existing
legacy nose tests to docker-compose.legacy.yml and update make targets to
support it.

FuhuXia committed:
remove CKAN 2.3 tests and functionality

FuhuXia committed:
Merge branch 'feature/py3' into feature/py3-ci

FuhuXia committed:
Merge branch 'feature/py3' into feature/docker-legacy

FuhuXia committed:
Merge pull request #95 from GSA/feature/py3-ci

New CI tests for CKAN 2.8 and 2.9

FuhuXia committed:
Merge pull request #96 from GSA/feature/docker-legacy

Update docker environment

FuhuXia committed:
copy legacy tests to /nose

also combined test-inventory/test_export.py in to /tests

FuhuXia committed:
rewrite nose tests to pytest

FuhuXia committed:
futurize to python 3

$ pip install futurize
$ futurize --both-stages --write ckanext

FuhuXia committed:
replace paste\|pylons.* import with toolkit

jbrown-xentity committed:
Update test_collections_ui tests to pass

jbrown-xentity committed:
Test cleanup

Clean up various test and configuration options

jbrown-xentity committed:
fix lint

jbrown-xentity committed:
ckan2.3 cleanup and linting

FuhuXia committed:
specify redis to be used for message queue

FuhuXia committed:
use postgres as db user for pg_isready

otherwise it used root and gives error

FuhuXia committed:
add ckanext.datajson.inventory_links_enabled into test.ini

somehow this is not working for us
config['ckanext.datajson.inventory_links_enabled'] = "True"

nickumia-reisys committed:
BaseController is only supported in PY2+CKAN2.9 (https://github.com/ckan/ckan/blob/2.9/ckan/lib/base.py); response doesn't seem to exist in ckan.plugins.toolkit, so import it from pylons again?

nickumia-reisys committed:
fixed lint; W293|E712|E722

nickumia-reisys committed:
fixed lint: E701

nickumia-reisys committed:
fixed lint: E703

nickumia-reisys committed:
fixed lint: E231

nickumia-reisys committed:
fixed lint: E401

nickumia-reisys committed:
fixed lint: E261

nickumia-reisys committed:
fixed lint: E111

nickumia-reisys committed:
fixed lint: E265

nickumia-reisys committed:
fixed lint: E302

nickumia-reisys committed:
majority of lint; but something in the tests are broken; I'm going to try and isolate the resets to a few files

nickumia-reisys committed:
finished all linting, except F403 and F405; tests work

nickumia-reisys committed:
fixed the unresolved import references; tests still work **wipes sweat**

nickumia-reisys committed:
pylons isn't py3 supported, so we can't revert response there.. ckan.plugins.toolkit should have response, but it is giving errors

jbrown-xentity committed:
Update factories to use meta

jbrown-xentity committed:
Handle python3 string with hash

jbrown-xentity committed:
Update setup and error assertions

jbrown-xentity committed:
Fix linting

nickumia-reisys committed:
made python-specific check so that the tests past both

nickumia-reisys committed:
IMPLEMENTED BLUEPRINTSgit status

nickumia-reisys committed:
fixed merge..

nickumia-reisys committed:
removed m.connects because itls not relevant to blueprints

nickumia-reisys committed:
updated blueprint to remove references to 'self', imported the correct new version of 'response' to handle 'zip' files

nickumia-reisys committed:
recommented the improper calls to 'Response.headers' because the code works without it; fixed lint

nickumia-reisys committed:
removed .coveragerc to allow setup.cfg to omit dirs from pytest properly; updated Makefile to point to the correct docker-compose.yml file

nickumia-reisys committed:
removed unused files (per James' request/review -- I agreed with him)

nickumia-reisys committed:
reset the return value for generate() in blueprint because no change was necessary; updated SQLAlchemy library calls in get_packages(..) because the package organization changed from 1.1.11 to 1.3.5

nickumia-reisys committed:
commented out test_export.py.. the test works in CKAN2.8, but the test framework was changed in CKAN2.9 and the response is forced to be a string, so the zip file can't be sent, see https://github.com/ckan/ckan/blob/2.9/ckan/tests/helpers.py#L185-L240 vs https://github.com/ckan/ckan/blob/2.8/ckan/tests/helpers.py#L147-L159

nickumia-reisys committed:
reduce pytest coverage because tests are failing

jbrown-xentity committed:
Use tee for output to shell and file

jbrown-xentity committed:
Only test on pull requests

nickumia-reisys committed:
updated makefile to have a conditional statement to use legacy docker-compose if older..

nickumia-reisys committed:
Merge branch 'feature/py3-py2-conditional' of github.com:GSA/ckanext-datajson into feature/py3-py2-conditional

nickumia-reisys committed:
fixed conditional statement because CKAN 2.8 is legacy and new..

nickumia-reisys committed:
testing without detach to see if any errors are thrown

nickumia-reisys committed:
reset circleci config and updated .env with old variable that was removed (maybe accidentally?)

nickumia-reisys committed:
trying to run legacy for CKAN 2.8

nickumia-reisys committed:
revert nose tests because they may have been accidentally updated with py3 stuff

nickumia-reisys committed:
fixed issue with environment variable CKAN__PLUGINS; incorrect syntax

jbrown-xentity committed:
Make test-legacy work with legacy docker

Ignore nose tests (pre py3) in linting

jbrown-xentity committed:
Exclude nose tests on github actions lint job

jbrown-xentity committed:
Remove test_collections_ui, is integration test

nickumia-reisys committed:
remove test_collections_ui because it will be re-implemented in catalog integration test

nickumia-reisys committed:
Merge branch 'feature/py3-py2-conditional' of github.com:GSA/ckanext-datajson into feature/py3-py2-conditional

jbrown-xentity committed:
Remove defunct circle-ci commands

jbrown-xentity committed:
Update code coverage threshold

jbrown-xentity committed:
remove defunct test

mogul committed:
Merge pull request #100 from GSA/feature/py3-py2-conditional

Feature/py3 py2 conditional

nickumia-reisys committed:
Add github actions badge

nickumia-reisys committed:
Merge pull request #101 from GSA/add-github-actions-badge

Add github actions badge

nickumia-reisys committed:
Link to github actions from github action badge

jbrown-xentity committed:
Add index for each package creation

jbrown-xentity committed:
Merge pull request #103 from GSA/feature/indexing

Add index for each package creation

nickumia-reisys committed:
new: update jquery to the correct filename

nickumia-reisys committed:
new: don't change the default mime_type for Response, just generate a unique Response instance with the desired mime_type

jbrown-xentity committed:
Merge pull request #102 from GSA/add-github-actions-badge-link

Link to github actions from github action badge

nickumia-reisys committed:
Merge pull request #104 from GSA/bugfix/jquery-path

new: update jquery to the correct filename

nickumia-reisys committed:
tests: ignore json files from zip test

nickumia-reisys committed:
new: fix for sub-agency issues

nickumia-reisys committed:
tests: overwrite CKANTestApp with custom CKANZipTestApp to allow binary zip files; create new test for sub-agencies part of a main organization

FuhuXia committed:
Merge pull request #106 from GSA/bugfix/inventory-doe-gsa-zip

Fix Inventory Errors - second attempt

FuhuXia committed:
split string for python2.7 and 3

nickumia-reisys committed:
Merge pull request #108 from GSA/string-split-py3

split string for python2.7 and 3

FuhuXia committed:
fix inventory_parent_uid

FuhuXia committed:
add config ckanext.datajson.export_map_filename

FuhuXia committed:
remove outdated readme

jbrown-xentity committed:
Merge pull request #110 from GSA/export-map-filename

add config ckanext.datajson.export_map_filename

jbrown-xentity committed:
Merge pull request #109 from GSA/inventory-parent-uid

fix inventory_parent_uid

FuhuXia committed:
validate email

FuhuXia committed:
go back to old faithful regex for email validation

nickumia-reisys committed:
Merge pull request #111 from GSA/validate-email

validate email

FuhuXia committed:
more regex for email validation

FuhuXia committed:
PEP 8

nickumia-reisys committed:
Merge pull request #113 from GSA/regex-email-validate

more regex for email validation

jguo144 commented on ckanext/datajson/blueprint.py:
```diff
@@ -0,0 +1,416 @@
+                # we didn't check ownership for this type of export, so never load private datasets here
+                packages = _get_ckan_datasets(org=owner_org)
+                if not packages:
+                    packages = get_packages(owner_org=owner_org, with_private=False)
```
This line should be commented out since we paginate the results in our fork. With this line the last pages will return all datasets.

We should also leave a comment explaining why it's commented out.

jguo144 commented on ckanext/datajson/blueprint.py:
```diff
@@ -0,0 +1,416 @@
+
+
+def _get_ckan_datasets(org=None, with_private=False):
+    n = 500
```
We can increase n to either 1000 or 2000

peterVorman replied:
done

FuhuXia committed:
Merge pull request #114 from GSA/str-encode

Str encode

FuhuXia committed:
remove encode to make writestr happy

otherwise it gives error:
TypeError("a bytes-like object is required, not 'str'")

FuhuXia committed:
parent could be None

This is one error that could lead to data.json export error

peterVorman committed:
improved codestile

updates after review

additional linters

jguo144 committed:
Merge remote-tracking branch 'origin/OD-1738/01-01-2022/Py3-support' into OD-1738/01-01-2022/Py3-support

jguo144 committed:
Fix lint

Fix lint issues. Ignore E402 for aliased imports from future package.

jguo144 committed:
Clean up after merge from upstream

jguo144 reviewed (approved):
LGTM
